UPX devel fails to build on macOS 11 (Big Sur)

## What's the problem (or question)?

Compiling UPX devel on macOS 11 Big Sur fails:

```
==> make all
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src all
Updating .depend
In file included from c_file.cpp:29:
./conf.h:37:6: error: "C++ 14 is required"
#    error "C++ 14 is required"
     ^
1 error generated.
```

(there are many more `C++ 14 is required` messages)

The system compiler does not have C++14 by default. `-std=c++14` needs to be passed.

## What should have happened?
The software should compile without having to specify extra flags.

## How can we reproduce the issue?
1. clone https://github.com/upx/upx.git
2. `make all`

## Please tell us details about your environment.
<!--- Include as many relevant details about the environment you experienced the bug in -->
* UPX version used (`upx --version`): `HEAD is now at e4d6eaf Tweak unoptimizeReloc again`
* Host Operating System and version: macOS 11.0.1
* Host CPU architecture: x86_64
* Target Operating System and version: macOS 11.0.1
* Target CPU architecture: x86_64

If I force it to use the right C++14 flags, I get more errors:

```
fx@rmeur src % make all CXXFLAGS=-std=c++14 -j1
clang++ '-DUPX_VERSION_GITREV="e4d6eafb9ef7"' -std=c++14 -o c_file.o -c c_file.cpp
In file included from c_file.cpp:29:
./conf.h:72:1: error: static_assert failed due to requirement '127 == 255' "CHAR_MAX == 255"
ACC_COMPILE_TIME_ASSERT_HEADER(CHAR_MAX == 255) // -funsigned-char
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./miniacc.h:2079:48: note: expanded from macro 'ACC_COMPILE_TIME_ASSERT_HEADER'
#    define ACC_COMPILE_TIME_ASSERT_HEADER(e)  static_assert(e, #e);
                                               ^             ~
In file included from c_file.cpp:29:
./conf.h:73:1: error: static_assert failed due to requirement '(char)(-1) > 0' "(char)(-1) > 0"
ACC_COMPILE_TIME_ASSERT_HEADER((char)(-1) > 0) // -funsigned-char
^                              ~~~~~~~~~~~~~~
./miniacc.h:2079:48: note: expanded from macro 'ACC_COMPILE_TIME_ASSERT_HEADER'
#    define ACC_COMPILE_TIME_ASSERT_HEADER(e)  static_assert(e, #e);
                                               ^             ~
2 errors generated.
```

--- a/src/Makefile
+++ b/src/Makefile
@@ -91,6 +91,7 @@ CXXFLAGS_WERROR ?= -Werror
 
 # Mandatory Flags - DO NOT CHANGE!
 mf.CPPFLAGS += $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES)
+mf.CXXFLAGS += -std=c++14
 mf.CXXFLAGS += $(CXXFLAGS_OPTIMIZE)
 # protect against security threats caused by misguided C++ compiler "optimizations"
 mf.CXXFLAGS += $(CXXFLAGS_NO_DELETE_NULL_POINTER_CHECKS)

